### PR TITLE
Update Spack submodule and new NCIO/MAPL versions

### PR DESCRIPTION
**Pull request by kgerheiser**
* Update MAPL to 2.22.0 which is used by ufs-weather-model
* Update NCIO to 1.1.2 which contains a bugfix
* Remove (LLVM) Clang tests for macOS. apple-clang is sufficent.
* Remove libjpeg-turbo from module blacklist and name it libjpeg.
* Update Spack submodule with latest jcsda_emc_spack_stack
* Update jedi-ufs-all with more specs

### CODE REVIEW

File: configs/common/packages.yaml [hunk 1]
```diff
@@ -111,7 +111,7 @@
       version: [1.6.37]
     # Attention - when updating also check discover site config
     mapl:
-      version: [2.12.3]
+      version: [2.20.0]
```
Comment (climbfuji) [inline configs/common/packages.yaml]: Hopefully this builds on macOS ...

**Review by climbfuji — APPROVED** [review]

File: configs/common/packages.yaml (hunk 1 above)
Comment (kgerheiser) [reply configs/common/packages.yaml]: Crap, this is the wrong version

Comment (climbfuji) [reply configs/common/packages.yaml]: Glad that I asked ;-) Let's see what the CI builds say.

**Comment by kgerheiser** [comment]
I removed libjpeg-turbo from the module blacklist. I was building NCEPLIBS-g2 using Spack dependencies and libjpeg should be available to end user.

**Comment by kgerheiser** [comment]
Something is wrong with the Clang build of MPI. C compiler cannot create executables.

**Comment by climbfuji** [comment]
> Something is wrong with the Clang build of MPI. C compiler cannot create executables.

That's new. Maybe they updated the macOS image and it's a different version now? Should we use a fixed version instead of LATEST?

**Comment by climbfuji** [comment]
> > Something is wrong with the Clang build of MPI. C compiler cannot create executables.
> 
> That's new. Maybe they updated the macOS image and it's a different version now? Should we use a fixed version instead of LATEST?

Also, we should probably be trying to do the correct two-level namespace change from Steve that we merged into spack rather than the half-baked solution that is in there now.

**Comment by kgerheiser** [comment]
This is the problem: https://github.com/actions/virtual-environments/issues/5846

**Comment by climbfuji** [comment]
> This is the problem: [actions/virtual-environments#5846](https://github.com/actions/virtual-environments/issues/5846)

Feel free to remove the (LLVM) clang tests (concretization, builds) entirely. It's enough in my opinion to have apple-clang and gcc.

**Comment by kgerheiser** [comment]
I was thinking that too

**Comment by kgerheiser** [comment]
Needed some fixes as noted here: https://github.com/NOAA-EMC/spack/pull/120

File: configs/common/modules.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ modules:
         ecmwf-atlas: '{compiler.name}/{compiler.version}/atlas/{version}'
         nlohmann-json: '{compiler.name}/{compiler.version}/json/{version}'
         nlohmann-json-schema-validator: '{compiler.name}/{compiler.version}/json-schema-validator/{version}'
+        libjpeg-turbo: '{compiler.name}/{compiler.version}/libjpeg/{version}'
```
Comment (climbfuji) [inline configs/common/modules.yaml:18]: This is only in the `tcl` section, I assume you want to do this for `lmod` as well?

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Review by climbfuji — APPROVED** [review]

**Pull request merged**